src/data/index.ts

```diff
@@ -7,19 +8,19 @@ import * as upload from "@src/data/upload";
 export { tag } from "@src/data/tag";
 
 export function start(): void {
-    upload.start();
-    target.reset();
     metric.start();
-    metadata.start();
-    ping.start();
-    tag.reset();
+    wrap(upload.start)();
+    wrap(target.reset)();
+    wrap(metadata.start)();
+    wrap(ping.start)();
+    wrap(tag.reset)();
 }
 
 export function end(): void {
-    tag.reset();
-    ping.end();
-    upload.end();
-    target.reset();
-    metadata.end();
+    wrap(tag.reset)();
+    wrap(ping.end)();
+    wrap(upload.end)();
+    wrap(target.reset)();
+    wrap(metadata.end)();
```

Comment:
>wrap [](start = 4, length = 4)

Minor; wrap is very general and doesn't tell you what it is doing. Why is it wrapped and what does the wrapper do? . Maybe time or measure or something a bit more specific.

Reply:
I kept it generic on purpose because eventually we may even want to introduce try / catch to eat exceptions if they happen. I looked at few other APIs and they go with generic wrap - so they can do few things at once. If you feel strongly about it, I will change the name. Please let me know.

Reply:
Renamed it to measure for now. When we need to do more, we will come back and call it wrap or something else later.